### PR TITLE
Mutable Metadata group name Part 3

### PR DESCRIPTION
**Pull request by cameronvoell**
This PR adds the Validation that group names obey the `update_group_name_policy` set upon creating a group.

See https://github.com/xmtp/libxmtp/issues/548 for more info.

Previous PR's:

- xmtp openmls: https://github.com/xmtp/openmls/pull/22
- proto: https://github.com/xmtp/proto/pull/158
- Part 1: https://github.com/xmtp/libxmtp/pull/643
- Part 2: https://github.com/xmtp/libxmtp/pull/644

PR's related to this change:

- xmtp openmls: https://github.com/xmtp/openmls/pull/23
- proto: https://github.com/xmtp/proto/pull/161
- proto: https://github.com/xmtp/proto/pull/171

Before merging this to `main`:
- [x] Update openmls reference in `Cargo.toml` to commit on `xmtp/openmls` `main` branch
- [x] Update dev/gen_protos.sh to point to `xmtp-proto` repo `main` branch

### CODE REVIEW

**Comment by cameronvoell** [comment]
I just noticed in the MLS spec that Commits are considered invalid if they have more than one Group Context Extensions Proposal (https://www.rfc-editor.org/rfc/rfc9420.html#section-12.2-3.8)

I'm going to take a closer look at what the conditions are that might lead to us having multiple proposals in a commit, and make sure we handle multiple GCE Proposals in the queue properly.

File: xmtp_mls/src/groups/group_permissions.rs [hunk 1]
```diff
@@ -294,29 +552,78 @@ impl PolicySet {
         })
     }
 
+    // In case group creator is on future version of libxmtp, we can validate
```
Comment (cameronvoell) [inline xmtp_mls/src/groups/group_permissions.rs:555]: This function enables users to validate metadata updates for metadata fields that might not exist in their version of libxmtp

File: xmtp_mls/src/groups/group_permissions.rs [hunk 2]
```diff
@@ -294,29 +552,78 @@ impl PolicySet {
...
+    // metadata policies on new unknown fields
+    fn evaluate_metadata_policy(&self, change: &MetadataChange, actor: &CommitParticipant) -> bool {
```
Comment (cameronvoell) [inline xmtp_mls/src/groups/group_permissions.rs:557]: This function enables users to validate metadata updates for metadata fields that might not exist in their version of libxmtp

**Comment by cameronvoell** [comment]
@neekolas I think this latest commit d9c26d52b2e5fbf175e6f285e7e54549ced2bcb6 incorporating the proto update to use mappings will achieve the functionality of allowing group members to validate metadata updates for fields that do not exist in their version of libxmtp. 

I'll spend some time today looking for spots to make the code cleaner, more intuitive, but any feedback is welcome 🙏

**Review by neekolas — APPROVED** [review]
This is looking great

**Pull request merged**